cluster-autoscaler/cloudprovider/huaweicloud/huaweicloud_auto_scaling_group.go

```diff
@@ -202,3 +226,29 @@ func (asg *AutoScalingGroup) Autoprovisioned() bool {
 func (asg *AutoScalingGroup) String() string {
 	return fmt.Sprintf("group: %s min=%d max=%d", asg.groupID, asg.minInstanceNumber, asg.maxInstanceNumber)
 }
+
+func (asg *AutoScalingGroup) deleteNodesFromCluster(nodeNames []string) error {
+	restConfig, err := clientcmd.BuildConfigFromFlags("", "")
+	if err != nil {
+		klog.Warningf("Failed to delete nodes from cluster due to can not get config. error: %v", err)
+		return err
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		klog.Warningf("Failed to delete nodes from cluster due to can not get kube-client. error: %v", err)
+		return err
+	}
+
+	// delete nodes one by one, return immediately when any error happens.
+	for _, nodeName := range nodeNames {
+		err := kubeClient.CoreV1().Nodes().Delete(context.TODO(), nodeName, metav1.DeleteOptions{})
+		if err != nil {
+			klog.Warningf("Failed to delete node from cluster. node: %s, error: %s", nodeName, err)
+			return err
```

Comment:
continue to delelte node when some error happen?  Instances is deleted, we need try our best to delete node record.

Reply:
That makes sense. Done.